sequencer/src/catchup.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{
+    api::endpoints::{AccountQueryData, BlocksFrontier},
+    state::{BlockMerkleCommitment, BlockMerkleTree, FeeAccount, FeeMerkleTree},
+};
+use hotshot_types::{data::ViewNumber, traits::node_implementation::ConsensusTime as _};
+use jf_primitives::merkle_tree::{MerkleCommitment as _, MerkleTreeScheme as _};
+use surf_disco::Client;
+use tide_disco::error::ServerError;
+use url::Url;
+
+#[derive(Debug, Clone, Default)]
+pub struct StatePeers {
+    clients: Vec<Client<ServerError>>,
+}
+
+impl StatePeers {
+    pub fn from_urls(urls: Vec<Url>) -> Self {
+        Self {
+            clients: urls.into_iter().map(Client::new).collect(),
+        }
+    }
+
+    pub async fn remember_account_balance(
+        &self,
+        view: ViewNumber,
+        account: FeeAccount,
+        fee_merkle_tree: &mut FeeMerkleTree,
+    ) {
+        loop {
+            for client in self.clients.iter() {
+                match client
+                    .get::<AccountQueryData>(&format!(
+                        "state/catchup/{}/account/{}",
+                        view.get_u64(),
+                        account.address()
+                    ))
+                    .send()
+                    .await
+                {
+                    Ok(res) => match res.proof.remember(fee_merkle_tree) {
+                        Ok(_) => return,
+                        Err(err) => tracing::warn!("Error verifying account proof: {}", err),
+                    },
+                    Err(err) => {
+                        tracing::warn!("Error fetching account from peer: {}", err);
+                    }
+                }
+            }
+            tracing::warn!("Could not fetch account from any peer, retrying");
+            // TODO: backoff? for testing a function that doesn't loop forever is probably more convenient.
+        }
+    }
+
+    pub async fn fetch_blocks_frontier(
+        &self,
+        view: ViewNumber,
+        root: BlockMerkleCommitment,
```

Comment:
We probably should also take the tree here and `remember` instead of `verify`. Even though the frontier is kind of special, forgetting and remembering it works basically the same as any other path.